dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,226 @@ else if( fieldbcontentname.startsWith("date") ){
-			try {
-				contentEditor = APILocator.getUserAPI().loadUserById(con.getModUser(),APILocator.getUserAPI().getSystemUser(),false);
-			} catch (Exception e1) {
-				Logger.error(ContentletAjax.class,e1.getMessage() + " no such user.  did mod_user get deleted?");
-				Logger.debug(ContentletAjax.class,e1.getMessage(), e1);
-				contentEditor = new User();
-			}
+					searchResult.put(fieldContentlet, fieldValue);
+				}
 
-			if (contentEditor.getFirstName() == null || contentEditor.getLastName() == null) {
-				user =con.getModUser();
-			} else {
-				user = contentEditor.getFullName();
-			}
-			PermissionAPI permissionAPI = APILocator.getPermissionAPI();
-			List<Permission> permissions = null;
-			try {
-				permissions = permissionAPI.getPermissions(con);
-			} catch (DotDataException e) {
-			}
-			StringBuffer permissionsSt = new StringBuffer();
-			Boolean ownerCanRead = false;
-			Boolean ownerCanWrite = false;
-			Boolean ownerCanPub = false;
-			for (Permission permission : permissions) {
-				String str = "P" + permission.getRoleId() + "." + permission.getPermission() + "P ";
-				if (permissionsSt.toString().indexOf(str) < 0) {
-					permissionsSt.append(str);
+				searchResult.put("inode", con.getInode());
+				searchResult.put("Identifier",con.getIdentifier());
+				searchResult.put("identifier", con.getIdentifier());
+				final Contentlet contentlet = con;
+				searchResult.put("__title__", conAPI.getName(contentlet, currentUser, false));
+
+				String spanClass = (s.getStructureType() ==1)
+						? "contentIcon"
+								:  (s.getStructureType() ==2)
+								? "gearIcon"
+										:  (s.getStructureType() ==3)
+										? "formIcon"
+											:  (s.getStructureType() ==4)
+											? "uknIcon " + UtilMethods.getFileExtension( ident.getURI()) + "Icon"
+												:  (s.getStructureType() ==5)
+												? "pageIcon"
+														: "personaIcon";
+
+				String typeStringToShow = s.getName() ;
+				searchResult.put("__type__", "<div class='typeCCol'><span class='" + spanClass +"'></span>&nbsp;" + typeStringToShow +"</div>");
+
+				String fieldValue = UtilMethods.dateToHTMLDate(con.getModDate()) + " " + UtilMethods.dateToHTMLTime(con.getModDate());
+
+				searchResult.put("modDate", fieldValue);
+				String user = "";
+				User contentEditor = null;
+				try {
+					contentEditor = APILocator.getUserAPI().loadUserById(con.getModUser(),APILocator.getUserAPI().getSystemUser(),false);
+				} catch (Exception e1) {
+					Logger.error(ContentletAjax.class,e1.getMessage() + " no such user.  did mod_user get deleted?");
+					Logger.debug(ContentletAjax.class,e1.getMessage(), e1);
+					contentEditor = new User();
 				}
+
+				if (contentEditor.getFirstName() == null || contentEditor.getLastName() == null) {
+					user = con.getModUser();
+				} else {
+					user = contentEditor.getFullName();
+				}
+				PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+				List<Permission> permissions = null;
 				try {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)